src/stdlib/test/double-conversion/cctest.h

```diff
@@ -0,0 +1,160 @@
+template<> inline void PrintfValue(unsigned short x) { printf("%hu", x); }
+template<> inline void PrintfValue(int64_t x) { printf("%" PRId64, x); }
+template<> inline void PrintfValue(uint64_t x) { printf("%" PRIu64, x); }
+template<> inline void PrintfValue(float x) { printf("%.30e", static_cast<double>(x)); }
+template<> inline void PrintfValue(double x) { printf("%.30e", x); }
+template<> inline void PrintfValue(bool x) { printf("%s", x ? "true" : "false"); }
+
+template<typename T1, typename T2>
+inline void CheckEqualsHelper(const char* file, int line,
+                       const char* expected_source,
+                       T1 expected,
+                       const char* value_source,
+                       T2 value) {
+  // If expected and value are NaNs then expected != value.
+  if (expected != value && (expected == expected || value == value)) {
+    printf("%s:%d:\n CHECK_EQ(%s, %s) failed\n",
+           file, line, expected_source, value_source);
+    printf("#  Expected: ");
+    PrintfValue(expected);
+    printf("\n");
+    printf("#  Found:    ");
+    PrintfValue(value);
+    printf("\n");
+    fflush(stdout);
+    abort();
+  }
+}
+
+template<>
+inline void CheckEqualsHelper(const char* file, int line,
+                       const char* expected_source,
+                       const char* expected,
+                       const char* value_source,
+                       const char* value) {
+  if ((expected == NULL && value != NULL) ||
+      (expected != NULL && value == NULL)) {
+    fflush(stdout);
+    abort();
+  }
+
+  if ((expected != NULL && value != NULL && strcmp(expected, value) != 0)) {
+    printf("%s:%d:\n CHECK_EQ(%s, %s) failed\n"
+           "#  Expected: %s\n"
+           "#  Found:    %s\n",
+           file, line, expected_source, value_source, expected, value);
+    fflush(stdout);
+    abort();
+  }
+}
+
+template<>
+inline void CheckEqualsHelper(const char* file, int line,
+                       const char* expected_source,
+                       const char* expected,
+                       const char* value_source,
+                       char* value) {
+  CheckEqualsHelper(file, line, expected_source, expected, value_source, static_cast<const char*>(value));
+}
+
+class CcTest {
+ public:
+  typedef void (TestFunction)();
+  CcTest(TestFunction* callback, const char* file, const char* name,
+         const char* dependency, bool enabled);
+  void Run() { callback_(); }
+  static int test_count();
+  static CcTest* last() { return last_; }
+  CcTest* prev() { return prev_; }
+  const char* file() const { return file_; }
+  const char* name() const { return name_; }
+  const char* dependency() const { return dependency_; }
+  bool enabled() const { return enabled_; }
+ private:
+  TestFunction* callback_;
+  const char* file_;
+  const char* name_;
+  const char* dependency_;
+  bool enabled_;
+  static CcTest* last_;
+  CcTest* prev_;
+};
+
+#endif  // ifndef CCTEST_H_
```

Comment:
For now, let's be a bit lazy and use the `cctest` test helper from https://github.com/google/double-conversion. In the long run, we should switch to https://github.com/catchorg/Catch2.